doc/rtd/explanation/format.rst

```diff
@@ -172,44 +172,8 @@ using a MIME archive.
 Part-handler
 ============
 
-This is a `part-handler`: It contains custom code for either supporting new
-mime-types in multi-part user data, or overriding the existing handlers for
-supported mime-types.  It will be written to a file in
-:file:`/var/lib/cloud/data` based on its filename (which is generated).
-
-This must be Python code that contains a ``list_types`` function and a
-``handle_part`` function. Once the section is read the ``list_types`` method
-will be called. It must return a list of mime-types that this `part-handler`
-handles. Since MIME parts are processed in order, a `part-handler` part
-must precede any parts with mime-types it is expected to handle in the same
-user data.
-
-The ``handle_part`` function must be defined like:
-
-.. code-block:: python
-
-    def handle_part(data, ctype, filename, payload):
-      # data = the cloudinit object
-      # ctype = "__begin__", "__end__", or the mime-type of the part that is being handled.
-      # filename = the filename of the part (or a generated filename if none is present in mime data)
-      # payload = the parts' content
-
-``Cloud-init`` will then call the ``handle_part`` function once before it
-handles any parts, once per part received, and once after all parts have been
-handled. The ``'__begin__'`` and ``'__end__'`` sentinels allow the part
-handler to do initialisation or teardown before or after receiving any parts.
-
-Begins with: ``#part-handler`` or ``Content-Type: text/part-handler`` when
-using a MIME archive.
-
-Example
--------
-
-.. literalinclude:: ../../examples/part-handler.txt
-   :language: python
-   :linenos:
-
-Also, `this blog post`_ offers another example for more advanced usage.
+This is a `part-handler`. It serves for implementing custom formats,
+see :ref:`custom_formats`.
```

Comment:
It feels a bit strange to obscure this text from this main format page because it also happens to be  custom formatting/extensions because we lose visibility to how a `text/part-handler` could be declared and provided as a user-data format. I think it makes this element a bit harder to discover for user-data types and I'd like to leave the bulk of this content here. Can leave this content in format.rst but cross link or a reference from the generated custom_modules.rst file, but leave this content homed in format.rst?

Reply:
I also think we should have a larger explanation here, but I'm ok keeping this as-is for now because of the bigger overhaul of this page in my PR.

Reply:
works for me. We can await that followup tomorrow which adds more structure to each section and may provide a simple separation between the two references to this material. Ideally I want to be sure that the user-data formats page at least has enough information to represent how the part-handler is declared as a user-data format type. Where the rest of the context/details live is up to your discretion once the second pass on this content provides a pass.

Reply:
https://github.com/canonical/cloud-init/pull/5548#issuecomment-2252327342